modules/core/classes/GalleryTemplateAdapter.class

```diff
@@ -870,16 +870,16 @@ class GalleryTemplateAdapter {
 	$buf = '';
 	if ($debug == 'buffered' || $profiling) {
 	    /* don't worry about localizing this -- it's only debug output */
-	    $buf .= '<div id="gpDebug"> <h3> Debug Output </h3>';
+	    $buf .= '<div id="gpDebug"> <h3> Debug Output </h3><pre>';
 	    if ($debug == 'buffered') {
 		$buf .= $gallery->getDebugBuffer();
 	    }
 
 	    if ($profiling) {
 		$storage =& $gallery->getStorage();
-		$buf .= '<span>' . $storage->getProfilingHtml() . '</span>';
+		$buf .= '</pre><span>' . $storage->getProfilingHtml() . '</span>';
 	    }
-	    $buf .= '</div>';
+	    $buf .= '</pre></div>';
```

Comment:
I think we only want to put the `</pre>` here if we didn't put it in the `if ($profiling)` case, right?

Reply:
Sure! In fact, I never tested this code, since it requires activating profiling.

So I gave profiling a try and I had to use an updated version of adodb temporaly (I think updating it to version 5 will be interesting) and I realize that profiling output is in HTML format, so I will refactor the `<pre>` tags a little bit and commit again.